Firestore/Source/API/FIRDocumentSnapshot.m

```diff
@@ -140,22 +140,26 @@ - (FIRSnapshotMetadata *)metadata {
                              options:[FSTFieldValueOptions optionsForSnapshotOptions:options]];
 }
 
-- (nullable id)valueForField:(id)field {
-  return [self valueForField:field options:[FIRSnapshotOptions defaultOptions]];
-}
-
-- (nullable id)valueForField:(id)field options:(FIRSnapshotOptions *)options {
+- (nullable FSTFieldValue*)fieldValueForKey:(id)key {
```

Comment:
This is confusingly similar to `valueForField:`. Also I'm not sure this name really works: "keys" generally refer to document keys but fields/field paths refer to fields within a document. This is using key to mean field.

If you renamed this `fieldValueForField` to match the behavior of the argument then I think the functional confusion is probably too much.

Really the most important thing to share here is the bit that converts a field argument (which is an NSString or FIRFieldPath) to an FSTFieldPath. An alternative arrangement that might be clearer would be to pull that part out into a separate `fieldPathForFieldArgument:` helper and then have both `valueForField` and your new method call that while leaving them both to actually index into `[[self.internalDocument data] valueForPath:]` for themselves.

Even after renaming this could be overcome instead with comments, perhaps contrasting how these behave or what their return types are, etc.